File: DSA/Khushi_590028673/Day4/Q1.cpp
// Q1 Transpose Matrix

#include <iostream>
#include <vector>

using namespace std;

vector<vector<int>> transpose(vector<vector<int>>& matrix) {
    int rows = matrix.size();
    int cols = matrix[0].size();

    vector<vector<int>> result(cols, vector<int>(rows));

    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            result[j][i] = matrix[i][j];
        }
    }

    return result;
}

int main() {
    int rows, cols;

    cout << "Enter number of rows and columns: ";
    cin >> rows >> cols;

    vector<vector<int>> matrix(rows, vector<int>(cols));

    cout << "Enter matrix elements:\n";
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            cin >> matrix[i][j];
        }
    }

    vector<vector<int>> transposed = transpose(matrix);

    cout << "\nTransposed Matrix:\n";
    for (const auto& row : transposed) {
        for (int val : row) {
            cout << val << " ";
        }
        cout << endl;
    }

    return 0;
}

File: DSA/vaishnavi_joshi_590016947/Day4/Q1.java
public class Transpose {
    public static void main(String[] args) {

        int[][] matrix = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };

        int rows = matrix.length;        
        int cols = matrix[0].length;    

        // Create transpose matrix
        int[][] transpose = new int[cols][rows];

        // Convert rows into columns
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                transpose[j][i] = matrix[i][j];
            }
        }


        System.out.println("Transposed Matrix:");

        for (int i = 0; i < cols; i++) {
            System.out.print("[");
            for (int j = 0; j < rows; j++) {
                System.out.print(transpose[i][j]);

                if (j < rows - 1) {
                    System.out.print(", ");
                }
            }
            System.out.println("]");
        }
    }
}